src/data/polynomial/partial_fractions.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2022 The Xena Project. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Sidharth Hariharan
+-/
+import ring_theory.localization.fraction_ring -- field of fractions
+import data.polynomial.div -- theory of division and remainder for monic polynomials
+import tactic.field_simp
+import tactic.linear_combination
+import data.zmod.basic
+import logic.function.basic
+
+/-!
+
+# Partial fractions
+
+These results were formalised by the Xena Project, at the suggestion
+of Patrick Massot.
+
+
+## The main theorems
+
+* General partial fraction decomposition theorem for polynomials over an integral domain R :
```

Comment:
Can you add the name of the main result?

Reply:
Hi, thank you for your suggestions! Incorporated all of them, except for the one about `div_eq_quo_add_rem_div` since I’m not sure where it should/could go…

As for the name of the result, as far as I know, it doesn’t really have a “name" (ie, I don’t think it’s referred to as “XYZ’s Theorem” for some XYZ)… maybe we can shorten it to “Partial fraction decomposition theorem”...

Reply:
I just mean the Lean declaration.

If my script is correct it can go in `ring_theory.integral_domain`, without adding any import.